Frameworks/CoreText/CTRun.mm

```diff
@@ -88,167 +72,158 @@ DWRITE_GLYPH_RUN __GetGlyphRunForDrawingInRange(const DWRITE_GLYPH_RUN& run, con
 @Status Interoperable
 */
 CFIndex CTRunGetGlyphCount(CTRunRef run) {
-    return run ? static_cast<_CTRun*>(run)->_dwriteGlyphRun.glyphCount : 0;
+    RETURN_RESULT_IF_NULL(run, 0);
+    return run->_dwriteGlyphRun.glyphCount;
 }
 
 /**
 @Status Interoperable
 */
 CFDictionaryRef CTRunGetAttributes(CTRunRef run) {
-    return run ? static_cast<CFDictionaryRef>(static_cast<_CTRun*>(run)->_attributes.get()) : nil;
+    RETURN_NULL_IF(!run);
+    return run->_attributes;
 }
 
 /**
  @Status Interoperable
- @Notes
 */
 CTRunStatus CTRunGetStatus(CTRunRef runRef) {
+    RETURN_RESULT_IF_NULL(runRef, kCTRunStatusNoStatus);
+
+	__CTRun* run = const_cast<__CTRun*>(runRef);
     CTRunStatus ret = kCTRunStatusNoStatus;
-    if (runRef) {
-        _CTRun* run = static_cast<_CTRun*>(runRef);
-
-        if (_GlyphRunIsRTL(run->_dwriteGlyphRun)) {
-            ret |= kCTRunStatusRightToLeft;
-            if (!std::is_sorted(run->_stringIndices.cbegin(), run->_stringIndices.cend(), std::greater<UINT16>())) {
-                ret |= kCTRunStatusNonMonotonic;
-            }
-        } else if (!std::is_sorted(run->_stringIndices.cbegin(), run->_stringIndices.cend())) {
+
+    if (_GlyphRunIsRTL(run->_dwriteGlyphRun)) {
+        ret |= kCTRunStatusRightToLeft;
+        if (!std::is_sorted(run->_stringIndices.cbegin(), run->_stringIndices.cend(), std::greater<UINT16>())) {
             ret |= kCTRunStatusNonMonotonic;
         }
+    } else if (!std::is_sorted(run->_stringIndices.cbegin(), run->_stringIndices.cend())) {
+        ret |= kCTRunStatusNonMonotonic;
+    }
 
-        if (run->_textMatrix != CGAffineTransformIdentity) {
-            ret |= kCTRunStatusHasNonIdentityMatrix;
-        }
+    if (run->_textMatrix != CGAffineTransformIdentity) {
+        ret |= kCTRunStatusHasNonIdentityMatrix;
     }
 
     return ret;
 }
 
 /**
  @Status Interoperable
- @Notes
 */
 const CGGlyph* CTRunGetGlyphsPtr(CTRunRef run) {
-    _CTRun* runPtr = static_cast<_CTRun*>(run);
-    return (runPtr && runPtr->_dwriteGlyphRun.glyphCount) ? runPtr->_dwriteGlyphRun.glyphIndices : nullptr;
+    RETURN_NULL_IF(!run);
```

Comment:
can do.

---
In reply to: [111453793](https://github.com/Microsoft/WinObjC/pull/2511#discussion_r111453793) [](ancestors = 111453793)